persistence/models/User.go

```diff
@@ -0,0 +1,35 @@
+package models
+
+import (
+	"fmt"
+	"net/mail"
+
+	"github.com/ditrit/badaas/services/auth/protocols/basicauth"
+)
+
+// Represents a user
+type User struct {
+	BaseModel
+	Username string `gorm:"not null"`
+	Email    string `gorm:"unique;not null"`
+	Password []byte
+}
+
+// Create a new user
+func NewUser(username, email, password string) (*User, error) {
```

Comment:
useless constructor, what you do in constructor have to be done in a `UserService` instead